src/background/services/paymentSession.ts

```diff
@@ -343,4 +278,57 @@ export class PaymentSession {
     this.amount = amount.toString()
     this.intervalInMs = Number((amount * BigInt(HOUR_MS)) / BigInt(this.rate))
   }
+
+  private async payContinuous() {
+    try {
+      const outgoingPayment =
+        await this.openPaymentsService.createOutgoingPayment({
+          walletAddress: this.sender,
+          incomingPaymentId: this.incomingPaymentUrl,
+          amount: this.amount
+        })
+      const { receiveAmount, receiver: incomingPayment } = outgoingPayment
+      const monetizationEventDetails: MonetizationEventDetails = {
+        amountSent: {
+          currency: receiveAmount.assetCode,
+          value: transformBalance(receiveAmount.value, receiveAmount.assetScale)
+        },
+        incomingPayment,
+        paymentPointer: this.receiver.id
+      }
+
+      sendMonetizationEvent({
+        tabId: this.tabId,
+        frameId: this.frameId,
+        payload: {
+          requestId: this.requestId,
+          details: monetizationEventDetails
+        }
+      })
+
+      // TO DO: find a better source of truth for deciding if overpaying is applicable
+      if (this.intervalInMs > 1000) {
+        this.tabState.saveOverpaying(this.tab, this.url, {
+          walletAddressId: this.receiver.id,
+          monetizationEvent: monetizationEventDetails,
+          intervalInMs: this.intervalInMs
+        })
+      }
+    } catch (e) {
+      if (isKeyRevokedError(e)) {
+        this.events.emit('open_payments.key_revoked')
+      } else if (isTokenExpiredError(e)) {
+        this.openPaymentsService.rotateToken()
+      } else if (e instanceof OpenPaymentsClientError) {
+        // We need better error handling.
+        if (e.status === 400) {
+          await this.setIncomingPaymentUrl()
```

Comment:
Should dedupe this call I think.

Reply:
I don't think it needs to be deduped, but we might have a small bug. If the `incomingPaymentUrl` is set, the `setIncomingPaymentUrl` is going to return only. If the incoming payment expires, a new one will not be created.